tests/functional/BackbeatConsumer.js

```diff
@@ -27,30 +26,53 @@ describe.skip('BackbeatConsumer', () => {
                 done();
             }
         }
-        producer = new BackbeatProducer({ zookeeper, topic });
-        consumer = new BackbeatConsumer({ zookeeper, groupId, topic,
-            queueProcessor });
+        producer = new BackbeatProducer({ kafka: kafkaConf, topic,
+                                          pollIntervalMs: 100 });
+        consumer = new BackbeatConsumer({ kafka: kafkaConf, groupId, topic,
+                                          queueProcessor });
         producer.on('ready', () => {
             producerReady = true;
             _doneIfReady();
         });
-        consumer.on('connect', () => {
+        consumer.on('ready', () => {
             consumerReady = true;
             _doneIfReady();
         });
     });
-    after(() => {
-        producer = null;
-        consumer = null;
-    });
-
-    it('should be able to read messages sent to the topic', done => {
-        producer.send(messages, () => {
-            consumer.subscribe().on('consumed', () => {
-                assert.deepStrictEqual(messages.map(e => e.message),
-                    consumedMessages);
+    after(done => {
+        producer.close(() => {
+            consumer.close(() => {
+                producer = null;
+                consumer = null;
                 done();
             });
         });
     });
+
+    it('should be able to read messages sent to the topic', done => {
+        let consumeCb = null;
+        let totalConsumed = 0;
+        async.series([
+            next => {
+                consumer.subscribe();
+                consumer.on('consumed', messagesConsumed => {
+                    totalConsumed += messagesConsumed;
+                    assert(totalConsumed <= messages.length);
+                    if (totalConsumed === messages.length) {
```

Comment:
If the assert above passes, will we ever enter this `if` statement block?

Reply:
It does. The assert doesn't throw an assertion when the condition is true. You can try this in your terminal.
```js
assert(5 <= 5);
if (5 === 5) {
   console.log('hello');
}
```